File: server/src/routes/projects.ts
import { Router } from 'express';
import { getCollection } from '../openproject/client.js';
import { mapProject } from '../openproject/mappers.js';
import { asyncHandler } from '../middleware/error.js';
import type { Project } from '../openproject/mappers.js';

const router = Router();

router.get(
  '/',
  asyncHandler(async (_req, res) => {
    const raw = await getCollection<Record<string, any>>('/api/v3/projects');
    const projects: Project[] = raw.map(mapProject);
    res.json(projects);
  }),
);

export default router;

File: server_py/app/routes/projects.py
from fastapi import APIRouter

from app.openproject.client import get_collection
from app.openproject.mappers import map_project
from app.openproject.schemas import Project


router = APIRouter(prefix="/api/projects")


@router.get("")
async def list_projects() -> list[Project]:
    raw = await get_collection("/api/v3/projects")
    return [map_project(item) for item in raw]
